sklearn/metrics/ranking.py

```diff
@@ -255,9 +266,33 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
                                         sample_weight=sample_weight)
         return auc(fpr, tpr, reorder=True)
 
-    return _average_binary_score(
-        _binary_roc_auc_score, y_true, y_score, average,
-        sample_weight=sample_weight)
+    if type_of_target(y_true) != "multiclass":
+        return _average_binary_score(
+            _binary_roc_auc_score, y_true, y_score, average,
+            sample_weight=sample_weight)
+    else:
+        # validation for multiclass parameter specifications
+        average_options = ("macro", "weighted")
+        if average not in average_options:
+            raise ValueError("average has to be one of {0}"
+                             "".format(average_options))
+        multiclass_options = ("ovo", "ovr")
+        if multiclass not in multiclass_options:
+            raise ValueError("{0} is not supported for multiclass ROC AUC"
+                             "".format(multiclass))
+
+        check_consistent_length(y_true, y_score)
+
+        if y_true.ndim == 1:
```

Comment:
there should be a call to `y_true = check_array(y_true)` before this to ensure that it is a numpy array before we check the `.ndim` attribute. See comment in PR about this.